### PR TITLE
Fix side-effect handling and allow custom errors in tests

### PR DESCRIPTION
**Pull request by pstutz**
Also make error message point to the problematic feature/scenario
 
Builds on top of 1c8fda6 and 6abd25e

### CODE REVIEW

File: tools/tck/src/main/scala/org/opencypher/tools/tck/api/CypherTCK.scala [hunk 1]
```diff
@@ -140,39 +144,50 @@ object CypherTCK {
...
-          List(ExpectError(errorType, time, detail).validate(), SideEffects().fillInZeros)
+          val expectedError = ExpectError(errorType, time, detail)
+          if (shouldValidate) {
+            expectedError.validate match {
+              case None => // No problem
+              case Some(errorMessage) =>
+                throw new InvalidFeatureFormatException(
+                  s"""Invalid error format in scenario "${pickle.getName}" from feature "$featureName":
+                    $errorMessage
+                    If this is a custom error, then disable this validation with tag "@allowCustomErrors"""")
```
Comment (Mats-SX) [inline tools/tck/src/main/scala/org/opencypher/tools/tck/api/CypherTCK.scala]: How are the custom errors validated?

**Comment by Mats-SX** [comment]
Rebased to get green build.

File: tools/tck/src/main/scala/org/opencypher/tools/tck/api/CypherTCK.scala (hunk 1 above)
Comment (pstutz) [reply tools/tck/src/main/scala/org/opencypher/tools/tck/api/CypherTCK.scala]: The flag disables error format validation.

**Comment by pstutz** [comment]
In the Neo4j acceptance tests there are errors that are not in the set of supported TCK errors. So we needed a flag to disable validation for them.

**Pull request merged**